app/src/main/java/com/cornellappdev/uplift/ui/components/gymdetail/GymAmenitySection.kt.kt

```diff
@@ -0,0 +1,143 @@
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.widthIn
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Icon
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.painter.Painter
+
+import androidx.compose.ui.res.colorResource
+
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import coil.compose.AsyncImagePainter.State.Empty.painter
+import com.cornellappdev.uplift.R
+import com.cornellappdev.uplift.fragment.GymFields
+import com.cornellappdev.uplift.models.EquipmentGrouping
+import com.cornellappdev.uplift.models.UpliftGym
+import com.cornellappdev.uplift.nav.navigateToClass
+import com.cornellappdev.uplift.util.GRAY01
+import com.cornellappdev.uplift.util.PRIMARY_BLACK
+import com.cornellappdev.uplift.util.montserratFamily
+
+/**
+ * Displays all of the Amenities that are included at the specific gym
+ */
+@OptIn(ExperimentalLayoutApi::class)
+@Composable
+fun GymAmenitySection(amenities: List<GymFields.Amenity?>?) {
+
+    if (amenities != null){
+        Spacer(modifier = Modifier.height(8.dp))
+        FlowRow(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(16.dp),
+            maxItemsInEachRow = 3,
+            horizontalArrangement = Arrangement.Center,
+            verticalArrangement = Arrangement.spacedBy(8.dp)
+
+
+        ) {
+            for(amenity in amenities){
+                if (amenity != null) {
+                    if(amenity.type.rawValue == "SHOWERS")
+                        AmenityElement(
+                            painter = painterResource(id = R.drawable.ic_shower),
+                            name = "Showers"
+                        )
+                    if(amenity.type.rawValue == "LOCKERS")
+                        AmenityElement(
+                            painter = painterResource(id = R.drawable.ic_lock),
+                            name = "Lockers"
+                        )
+                    if(amenity.type.rawValue == "PARKING")
+                        AmenityElement(
+                            painter = painterResource(id = R.drawable.ic_parking),
+                            name = "Parking"
+                        )
+                    if(amenity.type.rawValue == "ELEVATORS")
+                        AmenityElement(
+                            painter = painterResource(id = R.drawable.ic_elevator),
+                            name = "Elevators/Lifts"
+                        )
```

Comment:
There's a lot of code reuse here. Let's try to make this more concise.

Ideally, we only want one line calling `AmenityElement`. Maybe you can use a constant value `Map` from the amenity type to the painter resource id and the name, then just read the value of that map, passing `amenity.type.rawValue` as the key to that map.

Also, nitpick, but there are a ton of random newlines below this between the braces that you can probably touch up.